src/structs.py

```diff
@@ -1247,6 +1247,16 @@ class DemonstrationResponse(Response):
     teacher_traj: Optional[LowLevelTrajectory]
 
 
+@dataclass(frozen=True, eq=False, repr=False)
+class StateBasedDemonstrationQuery(Query):
```

Comment:
i think we should rename `DemonstrationQuery` since both `DemonstrationQuery` and `StateBasedDemonstrationQuery` are currently using `DemonstrationResponse`

Reply:
also move this class up above the associated response?